proseco/jupiter.py

```diff
@@ -0,0 +1,151 @@
+import astropy.units as u
+import numpy as np
+from astropy.table import Table
+from chandra_aca import planets
+from chandra_aca.transform import eci_to_radec, radec_to_yagzag, yagzag_to_pixels
+from cheta.comps import ephem_stk
+from cxotime import CxoTime
+
+
+def get_jupiter_position(date, duration, att):
+    date0 = CxoTime(date)
+    n_times = int(duration / 1000) + 1
+    dates = date0 + np.linspace(0, duration, n_times) * u.s
+    times = np.atleast_1d(dates.secs)
+
+    chandra_ephem = ephem_stk.get_ephemeris_stk(start=dates[0], stop=dates[-1])
+
+    # Get Jupiter positions using chandra_aca.planets
+    ephem = {
+        key: np.interp(times, chandra_ephem["time"], chandra_ephem[key])
+        for key in ["x", "y", "z"]
+    }
+    pos_earth = planets.get_planet_barycentric("earth", dates)
+
+    chandra_eci = np.zeros_like(pos_earth)
+    chandra_eci[..., 0] = ephem["x"].reshape(times.shape) / 1000
+    chandra_eci[..., 1] = ephem["y"].reshape(times.shape) / 1000
+    chandra_eci[..., 2] = ephem["z"].reshape(times.shape) / 1000
+    eci = planets.get_planet_eci("jupiter", dates, pos_observer=pos_earth + chandra_eci)
+
+    # Convert to RA, Dec
+    ra, dec = eci_to_radec(eci)
+    yag, zag = radec_to_yagzag(ra, dec, att)
+    row, col = yagzag_to_pixels(yag, zag, allow_bad=True)
+
+    lim0 = -512
+    lim1 = 511
+    # Only care about planet within the CCD
+    ok = (row >= lim0) & (row <= lim1) & (col >= lim0) & (col <= lim1)
+    if np.any(ok):
+        out = Table(
+            {
+                "times": times[ok],
+                "rows": row[ok],
+                "cols": col[ok],
+            }
+        )
+    else:
+        out = None
+    return out
+
+
+def jupiter_distribution_check(cand_guide_set, jupiter_data):
+    """
+    Check that there are at least two candidate guide stars on the side of the CCD
+    that does not have Jupiter.
+
+    :param cand_guide_set: Table of candidate guide stars
+    :returns: bool (True if check passes)
+    """
+    sign_max = np.sign(np.max(jupiter_data["rows"]))
```

Comment:
I think I want to add a little padding to the check such that Jupiter very close to the midline counts on both sides of the CCD anyway.